ee/secureenclavesigner/secureenclavesigner_darwin.go

```diff
@@ -0,0 +1,240 @@
+//go:build darwin
+// +build darwin
+
+package secureenclavesigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"os/user"
+	"strings"
+	"sync"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/launcher/ee/allowedcmd"
+	"github.com/kolide/launcher/ee/consoleuser"
+)
+
+const (
+	CreateKeyCmd = "create-key"
+)
+
+type opt func(*SecureEnclaveSigner)
+
+type SecureEnclaveSigner struct {
+	uidPubKeyMap         map[string]*ecdsa.PublicKey
+	pathToLauncherBinary string
+	persister            persister
+	slogger              *slog.Logger
+	mux                  *sync.Mutex
+}
+
+// persister is an interface that allows the secure enclave signer to persist
+// keys it creates, this is needed since we may need to create a new key anytime
+// a new user logs in on macos
+type persister interface {
+	Persist([]byte) error
+}
+
+func New(slogger *slog.Logger, persister persister, opts ...opt) (*SecureEnclaveSigner, error) {
+	ses := &SecureEnclaveSigner{
+		uidPubKeyMap: make(map[string]*ecdsa.PublicKey),
+		persister:    persister,
+		slogger:      slogger.With("component", "secureenclavesigner"),
+		mux:          &sync.Mutex{},
+	}
+
+	for _, opt := range opts {
+		opt(ses)
+	}
+
+	if ses.pathToLauncherBinary == "" {
+		p, err := os.Executable()
+		if err != nil {
+			return nil, fmt.Errorf("getting path to launcher binary: %w", err)
+		}
+
+		ses.pathToLauncherBinary = p
+	}
+
+	return ses, nil
```

Comment:
wonder if we should do some kind of key verification here? we can at least query secure enclave to make sure it recognizes the users key.

Reply:
ehh .. not sure it's worth the extra execs

we don't save a key unless secure enclave creates it
once we store the key server side, if the key is lost some how launcher would need a new install if were strict on validating hardware